Statically compiled Perl 5.12 cannot load POSIX, thus we inline
this code from bits/wexitstatus.h.

diff -aur autoconf-2.68.orig/lib/Autom4te/FileUtils.pm autoconf-2.68/lib/Autom4te/FileUtils.pm
--- autoconf-2.68.orig/lib/Autom4te/FileUtils.pm	2011-04-12 17:07:09.623166965 +0000
+++ autoconf-2.68/lib/Autom4te/FileUtils.pm	2011-04-12 17:11:26.056424914 +0000
@@ -263,20 +263,18 @@
     }
   else
     {
-      use POSIX qw (WIFEXITED WEXITSTATUS WIFSIGNALED WTERMSIG);
-
-      if (WIFEXITED ($?))
+      if (!($? & 0x7f))
 	{
-	  my $status = WEXITSTATUS ($?);
+	  my $status = ($? & 0xff00) >> 8;
 	  # Propagate exit codes.
 	  fatal ('',
 		 "$command failed with exit status: $status",
 		 exit_code => $status)
 	    unless $status == $expected;
 	}
-      elsif (WIFSIGNALED ($?))
+      elsif (((($? & 0x7f) + 1) >> 1) > 0)
 	{
-	  my $signal = WTERMSIG ($?);
+	  my $signal = $? & 0x7f;
 	  fatal "$command terminated by signal: $signal";
 	}
       else
diff -aur autoconf-2.68.orig/lib/Autom4te/General.pm autoconf-2.68/lib/Autom4te/General.pm
--- autoconf-2.68.orig/lib/Autom4te/General.pm	2011-04-12 17:07:09.623166965 +0000
+++ autoconf-2.68/lib/Autom4te/General.pm	2011-04-12 17:13:56.303136721 +0000
@@ -387,19 +387,17 @@
     }
   else
     {
-      use POSIX qw (WIFEXITED WEXITSTATUS WIFSIGNALED WTERMSIG);
-
-      if (WIFEXITED ($?))
+      if (!($? & 0x7f))
 	{
-	  my $status = WEXITSTATUS ($?);
+          my $status = ($? & 0xff00) >> 8;                   
 	  # WIFEXITED and WEXITSTATUS can alter $!, reset it so that
 	  # error() actually propagates the command's exit status, not $!.
 	  $! = 0;
 	  error "$command failed with exit status: $status";
 	}
-      elsif (WIFSIGNALED ($?))
+      elsif (((($? & 0x7f) + 1) >> 1) > 0)                 
 	{
-	  my $signal = WTERMSIG ($?);
+          my $signal = $? & 0x7f;                                         
 	  # In this case we prefer to exit with status 1.
 	  $! = 1;
 	  error "$command terminated by signal: $signal";
